ethcore/src/blockchain/blockchain.rs

```diff
@@ -665,6 +665,21 @@ impl BlockChain {
 		self.db.key_value().read_with_cache(db::COL_EXTRA, &self.block_details, parent).map_or(false, |d| d.children.contains(hash))
 	}
 
+	/// fetches the list of blocks from best block to n, and n's parent hash
+	/// where n > 0
+	pub fn block_headers_from_best_block(&self, n: u64) -> Option<(Vec<encoded::Header>, H256)> {
+		let mut blocks = vec![];
```

Comment:
Because we know that the number blocks, a possible to optimization could be to allocate only once

```suggestion
		let mut blocks = Vec::with_capacity(n);
```